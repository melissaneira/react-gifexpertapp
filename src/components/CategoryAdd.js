import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = ({setCategories}) => {
  const [inputValue, setInputValue] = useState('')
  const handleInputChange = (e) =>{
    //console.log(e.target.value);
    setInputValue(e.target.value);
    //console.log('handleInputChange llamado');
  }

  const handleSubmit=(e) => {
      e.preventDefault();
      if(inputValue.trim().length > 2){
        setCategories(cats => [ inputValue, ...cats] );
        setInputValue("");
      }
      
      console.log('handle Submit', inputValue);
  }
  return (
     <form onSubmit={handleSubmit}>
       <p>{inputValue}</p>
        <input 
            type="text"
            value= {inputValue}
            onChange= {handleInputChange}
            ></input>
        
    </form>

  )
}

CategoryAdd.propTypes = {
  setCategories : PropTypes.func.isRequired
}

export default CategoryAdd