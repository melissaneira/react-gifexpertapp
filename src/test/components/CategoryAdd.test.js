import { shallow } from "enzyme"
import CategoryAdd from "../../components/CategoryAdd";
import '@testing-library/jest-dom';


describe('Pruebas en <CategoryAdd />', () => { 
    const setCategories = jest.fn();
    let wrapper;
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<CategoryAdd setCategories={setCategories}/>);        
    });

    test('Debe rederizar el componente CategoryAdd', () => {         
        expect(wrapper).toMatchSnapshot();        
    })    

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target : {value}});
        const p = wrapper.find('p');
        expect(p.text()).toBe(value);
     })

     test('no debe de postear la informacion con submit', () => { 
         wrapper.find('form').simulate('submit', {preventDefault(){}});
         expect(setCategories).not.toHaveBeenCalled();
      })

    test('debe de llamr el setCategories y limpiar la caja de texto', () => { 
        let input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target : {value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toBeCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        input = wrapper.find('input');
        expect(input.prop('value')).toBe('');
    })
})