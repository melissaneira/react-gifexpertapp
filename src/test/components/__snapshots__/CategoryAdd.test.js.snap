// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CategoryAdd /> Debe rederizar el componente CategoryAdd 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
